kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp

```diff
@@ -306,6 +307,30 @@ namespace Kratos
             KRATOS_CHECK_LESS_EQUAL(std::abs((rA(4,4) - 2069000000.000000000)/rA(4,4)), tolerance);
             KRATOS_CHECK_LESS_EQUAL(std::abs((rA(5,5) - 1.000000000)/rA(5,5)), tolerance);
 
+            // Testing scale
+            BuilderAndSolverType::Pointer p_builder_and_solver_scale = BuilderAndSolverType::Pointer( new ResidualBasedBlockBuilderAndSolverType(p_solver, true) );
+            
+            const SparseSpaceType::MatrixType& rA_scale = BuildSystem(r_model_part, p_scheme, p_builder_and_solver_scale);
+            
+            KRATOS_CHECK(rA.size1() == 6);
+            KRATOS_CHECK(rA.size2() == 6);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA_scale(0,0) - 2069000000.000000000)/rA(0,0)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA_scale(1,1) - 2.26648e+10)/rA_scale(1,1)), 1.0e-5);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA_scale(2,2) - 4138000000.000000000)/rA(2,2)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA_scale(2,4) - -2069000000.000000000)/rA(2,4)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA_scale(3,3) - 2.26648e+10)/rA_scale(3,3)), 1.0e-5);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA_scale(4,2) - -2069000000.000000000)/rA(4,2)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA_scale(4,4) - 2069000000.000000000)/rA(4,4)), tolerance);
+            KRATOS_CHECK_LESS_EQUAL(std::abs((rA_scale(5,5) - 2.26648e+10)/rA_scale(5,5)), 1.0e-5);
+            
+            SparseSpaceType::MatrixType copy_A(rA);
+            const double condition_number_not_scale = ConditionNumberUtility().GetConditionNumber(copy_A);
+            SparseSpaceType::MatrixType copy_A_scale(rA_scale);
+            const double condition_number_scale = ConditionNumberUtility().GetConditionNumber(copy_A_scale);
+
+            KRATOS_CHECK_RELATIVE_NEAR(condition_number_not_scale, 5.41671e+09, 1.0e-5);
```

Comment:
here you can seeile the condition number is reduced in 8 orders!!!!

Reply:
@KratosMultiphysics/implementation-committee the tests shows a reduction in the condition number of 8 orders